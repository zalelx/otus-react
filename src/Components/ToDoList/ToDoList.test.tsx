import {Task, ToDoList} from "./index";
import {shallow, ShallowWrapper} from "enzyme";
import React from "react";
import {matchers} from "jest-emotion";

let wrapper: ShallowWrapper<React.Component["props"],
    React.Component["state"],
    ToDoList>;

expect.extend(matchers);

const fetchTasks = jest.fn(() => {
    return new Promise<Task[]>((resolve) =>
        resolve([{checked: true, text: "asfd"}])
    );
});
const updateTasks = jest.fn(() => {
    return new Promise<void>((resolve) => resolve());
});

beforeEach(() => {
    wrapper = shallow(
        <ToDoList updateTasks={updateTasks} fetchTasks={fetchTasks} />
    );
});

describe("<Components.ToDoList/> rendering", () => {
    it("renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("should have add button", () => {
        expect(wrapper.find("button")).toHaveLength(1);
    });
});
