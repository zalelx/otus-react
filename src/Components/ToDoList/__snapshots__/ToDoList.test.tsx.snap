// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Components.ToDoList/> rendering renders correctly 1`] = `
<Fragment>
  <div>
    <ToDoItem
      checked={true}
      key="0"
      onCheck={[Function]}
      onRemove={[Function]}
      task="asfd"
    />
  </div>
  <button
    onClick={[Function]}
  >
    Add
  </button>
</Fragment>
`;
